### PR TITLE
Refactor Android payload configuration into a byte array

### PR DESCRIPTION
**Pull request by timwr**
See framework pull request.
The removes some code duplication (e.g PayloadTrustManager) by adding java code that is shared between the Android payload and meterpreter.

### CODE REVIEW

File: java/androidpayload/app/src/com/metasploit/stage/Payload.java [hunk 1]
```diff
@@ -19,17 +19,19 @@
...
-    public static final String CERT_HASH =      "WWWW                                        ";
-    public static final String TIMEOUTS =       "TTTT                                                ";
+    public static final byte [] configBytes = new byte[] {
+            (byte) 0xde, (byte) 0xad, (byte) 0xba, (byte) 0xad, //placeholder
```
Comment (bcook-r7) [inline java/androidpayload/app/src/com/metasploit/stage/Payload.java]: seriously Java :| (I do admire your dedication here :)

**Comment by bcook-r7** [comment]
I merged to test out our builder, ran into this:
```
23:58:35 -------------------------------------------------------
23:58:35  T E S T S
23:58:35 -------------------------------------------------------
23:58:35 Running metasploit.PayloadTest
23:58:38 Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.822 sec
23:58:38 Running javapayload.stage.MeterpreterTest
23:58:38 Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
23:58:38 Running javapayload.stage.ShellTest
23:58:38 Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.007 sec <<< FAILURE!
23:58:38 
23:58:38 Results :
23:58:38 
23:58:38 Failed tests:   testShellStage(javapayload.stage.ShellTest): MagicToken missing in shell output: /bin/sh: 2: exit
: not found
23:58:38 
23:58:38 Tests run: 7, Failures: 1, Errors: 0, Skipped: 0
```

**Comment by bcook-r7** [comment]
So tests need to be updated to check for the new token, or something else need to be updated.

**Comment by bcook-r7** [comment]
Sigh, I'm not reproing the test failure outside of that build environment (or would have seen it locally). Investigating...

**Comment by timwr** [comment]
Oops I'm seeing this on non-injected (-x) payloads:
```
W/dalvikvm(24085): Class resolved by unexpected DEX: Lcom/metasploit/meterpreter/Meterpreter;(0x42e1fec8):0x75893000 ref [Lcom/metasploit/stage/ConfigParser;] Lcom/metasploit/stage/ConfigParser;(0x42a28060):0x72f8d000
W/dalvikvm(24085): (Lcom/metasploit/meterpreter/Meterpreter; had used a different Lcom/metasploit/stage/ConfigParser; during pre-verification)
W/System.err(24085): java.lang.reflect.InvocationTargetException
W/System.err(24085): 	at java.lang.reflect.Method.invokeNative(Native Method)
W/System.err(24085): 	at java.lang.reflect.Method.invoke(Method.java:525)
W/System.err(24085): 	at com.metasploit.stage.Payload.runNextStage(Payload.java:198)
W/System.err(24085): 	at com.metasploit.stage.Payload.runStageFromHTTP(Payload.java:132)
W/System.err(24085): 	at com.metasploit.stage.Payload.main(Payload.java:103)
W/System.err(24085): 	at com.metasploit.stage.Payload$1.run(Payload.java:45)
W/System.err(24085): Caused by: java.lang.reflect.InvocationTargetException
W/System.err(24085): 	at java.lang.reflect.Constructor.constructNative(Native Method)
W/System.err(24085): 	at java.lang.reflect.Constructor.newInstance(Constructor.java:417)
W/System.err(24085): 	at androidpayload.stage.Meterpreter.start(Meterpreter.java:43)
W/System.err(24085): 	... 6 more
W/System.err(24085): Caused by: java.lang.IllegalAccessError: Class ref in pre-verified class resolved to unexpected implementation
W/System.err(24085): 	at com.metasploit.meterpreter.Meterpreter.loadConfiguration(Unknown Source)
W/System.err(24085): 	at com.metasploit.meterpreter.AndroidMeterpreter.<init>(AndroidMeterpreter.java:136)
W/System.err(24085): 	... 9 more
```
One problem I was trying to solve with this PR is that there is no shared code between the Payload and the Meterpreter (and the java Payload, but that's another story), so you end up with 3 classes sometimes (e.g PayloadTrustManager). 
Maybe it's best we revert this while I think of a better solution, e.g:
- Metaprogramming (not great for java)
- Obfuscating the Payload with proguard 
- Revert and put up with the code duplication for now (PR incoming).

Apologies!

**Comment by busterb** [comment]
No worries, it never made it out of testing in the end.

**Pull request merged**